dev/AppNotifications/AppNotificationContentBuilder/AppNotificationContentBuilder.idl

```diff
@@ -0,0 +1,148 @@
+﻿import "oaidl.idl";
+import "ocidl.idl";
+
+[contract(AppNotificationsContract, 1)]
+runtimeclass AppNotificationButtonBuilder
+{
+    Constructors
+    // Initializes a new instance of the ToastButton class.
+    ToastButton()
+
+    // Initializes a new instance of the ToastButton class.
+    ToastButton(String, String)
+
+
+    Properties
+    // Gets or sets additional options relating to activation of the toast button.New in Creators Update
+    ActivationOptions
+
+    // Gets or sets what type of activation this button will use when clicked.Defaults to Foreground.
+    ActivationType
+
+    // Gets app - defined string of arguments that the app can later retrieve once it is activated when the user clicks the button.Required
+    Arguments
```

Comment:
If this is required should it also be a setter?

Reply:
Setters in a fluent API are problematic since they break the chain of function calls. Getters less so because they shouldn't be used in the call chain.